### ISSUE TITLE
NullReferenceException when HTML payload ends with a comment

### ISSUE THREAD

**Issue by enkelmedia** [AngleSharp/AngleSharp#459]
When I'm parsing this code I get an error.

```html
<html xmlns="http://www.w3.org/1999/xhtml">
<head></head>
<body></body>
</html>
<!-- Comment -->
```

```plain
System.NullReferenceException: Object reference not set to an instance of an object.\r\n   

at AngleSharp.AutoSelectedMarkupFormatter.Comment(IComment comment)\r\n   at AngleSharp.Dom.Comment.ToHtml(TextWriter writer, IMarkupFormatter formatter)\r\n   at AngleSharp.Dom.Collections.NodeList.ToHtml(TextWriter writer, IMarkupFormatter formatter)\r\n   

at AngleSharp.Extensions.FormatExtensions.ToHtml(IMarkupFormattable markup, IMarkupFormatter formatter)\r\n   at PreMailer.Net.PreMailer.MoveCssInline(Boolean removeStyleElements, String ignoreElements, String css, Boolean stripIdAndClassAttributes, Boolean removeComments)\r\n   
```

Removing the comment in the end solves the issue.

```html
<html xmlns="http://www.w3.org/1999/xhtml">
<head></head>
<body></body>
</html>
```

**Comment by FlorianRappl** [comment]
From the dump you've shown it does not seem that parsing is the problem - rather the serialization. Can you confirm and show the code (not the document, i.e., present a MWE) that crashes?

Thanks!

**Comment by enkelmedia** [comment]
I think that it might be when it's converted to a HtmlDocument?

**Comment by FlorianRappl** [comment]
Nope - there is no such process. It starts with an `HtmlDocument` and ends with an `HtmlDocument`. MWE to reproduce?

**Comment by FlorianRappl** [comment]
Which version are you using? I can't reproduce the bug.

**Comment by enkelmedia** [comment]
I'm using the latest version - but trough PreMailer.NET. Since the exception was trown from AngleSharp i figured that the issues was there. If my example with the comment works then the issue is in PreMailer.NET - sorry for that!

**Comment by FlorianRappl** [comment]
I assume the error is in [this line](https://github.com/milkshakesoftware/PreMailer.Net/blob/9e99d2c56f20b391186764774c2d807b34a15d7c/PreMailer.Net/PreMailer.Net/PreMailer.cs#L123).

I've checked with the current code and everything seems to be fine here. My guess is that the problem indeed existed (and has to do with comments in general - independent of their position!), but is already gone due to some fix earlier in the process of v0.10.

**Comment by DaniilVeriga** [comment]
There is the same issue with a comment in the beginning: #610
